### DISCUSSION TITLE
Does MICOM no longer calculate shadow_prices and reduced_costs?

### DISCUSSION THREAD

**Discussion by ChristianLieven** [micom-dev/micom#45] [Q&A] [answered]
Hey @cdiener,

I've been enjoying working with MICOM for the past year now but it struck me as odd that I find `shadow_prices` and `reduced_costs` mentioned in the docstring of the `CommunitySolution` but no further use anywhere? 

Was MICOM able to generate/ output those at one point until the feature was removed, or is this a planned feature? 

How would I best go about calculating those two metrics for a given solution? How do the `micom.elasticity.elasticities` relate to them?

Best regards,

Christian

**Comment by cdiener** [comment, marked as answer]
Oh yeah, that was dropped very early. The reason was that getting the solution for everything into Python is the slowest part of solving the model. So dropping those makes getting the solution takes about 1/3 of the time. Also due to two-step optimization is a bit unclear what those would correspond to (the community objective or the QP problem in step 2). If you would add them back in now it would give you the values for the QP problem which would be related to the impact of fluxes on the regularization. 

The elasticities are a normalized intervention effect. It increases the import bound for each imported metabolite and the abundances for each microbe (each one individually) and tracks the effect on the target fluxes, so it is a measure of (dv/v) / (dp/p) for each flux and input parameter (import bounds or microbe abundances). The name might have been poorly chosen. I think they are more like response coefficients in the classical setting. I wanted to try to get an analytical solution for those at some point but haven't gotten to it.
